### PR TITLE
marathon and chronos proxy minions

### PR DESCRIPTION
**Pull request by toddtomkinson**
This patch includes support for treating marathon and chronos clusters as minions via a proxy minion.

I demoed this at the salt meetup in Lehi on October 1st (sorry it took so long to clean things up).

Please let me know if anything needs to be updated or if there are other utilities that should be used.

### CODE REVIEW

**Comment by toddtomkinson** [comment]
@cro, this should work with the latest proxy minion updates.  I've verified things work at least with the changes on the 2015.8 branch.

**Comment by toddtomkinson** [comment]
I'm fixing the lint errors now--an update will be coming shortly.

**Comment by cro** [comment]
Good deal, thanks.

On Wed, Nov 11, 2015 at 11:31 AM, Todd Tomkinson <notifications@github.com>
wrote:

> I'm fixing the lint errors now--an update will be coming shortly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saltstack/salt/pull/28778#issuecomment-155869333>.
>



-- 
*--cro*
C. R. Oldham, Platform Engineer, SaltStack
cr@saltstack.com / 801-564-4673

File: salt/modules/chronos.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+'''
+Module providing a simple management interface to a chronos cluster.
+
+Currently this only works when run through a proxy minion.
+
+.. versionadded:: 2015.8.2
```
Comment (jfindlay) [inline salt/modules/chronos.py]: @toddtomkinson, this should be `.. versionadded:: Boron`.

File: salt/modules/marathon.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+'''
+Module providing a simple management interface to a marathon cluster.
+
+Currently this only works when run through a proxy minion.
+
+.. versionadded:: 2015.8.2
```
Comment (jfindlay) [inline salt/modules/marathon.py]: @toddtomkinson, same here.

File: salt/proxy/chronos.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+'''
+Chronos
+========
+
+Proxy minion for managing a Chronos cluster.
+
```
Comment (jfindlay) [inline salt/proxy/chronos.py]: @toddtomkinson, will you also add a `.. versionadded:: Boron` here?

File: salt/proxy/marathon.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+'''
+Marathon
+========
+
```
Comment (jfindlay) [inline salt/proxy/marathon.py]: @toddtomkinson, `.. versionadded:: Boron`.

File: salt/states/chronos_job.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+'''
+Configure Chronos jobs via a salt proxy.
+
```
Comment (jfindlay) [inline salt/states/chronos_job.py]: @toddtomkinson, `.. versionadded:: Boron`.

File: salt/states/marathon_app.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+'''
+Configure Marathon apps via a salt proxy.
+
```
Comment (jfindlay) [inline salt/states/marathon_app.py]: @toddtomkinson, `.. versionadded:: Boron`.

File: salt/modules/chronos.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    jobs = _jobs()
+    if name in jobs:
+        return {'job': jobs[name]}
+    return None
+
+
+def update_job(name, config):
+    '''
+    Update the specified job with the given configuration.
+    '''
```
Comment (jfindlay) [inline salt/modules/chronos.py]: This needs a CLI Example.

File: salt/modules/marathon.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        decode_type='json',
+        decode=True,
+    )
+    return response['dict']
+
+
+def update_app(id, config):
+    '''
+    Update the specified app with the given configuration.
+    '''
```
Comment (jfindlay) [inline salt/modules/marathon.py]: @toddtomkinson, `CLI Example` needed here.

**Comment by jfindlay** [comment]
@toddtomkinson, this is awesome, thanks for the pull request.

**Comment by toddtomkinson** [comment]
@jfindlay, do I need to add the versionadded on the grains, too?

**Comment by jfindlay** [comment]
@toddtomkinson, yeah that would be good, thanks.

**Comment by cro** [comment]
@toddtomkinson Apologies, I merged this before I realized you hadn't added the ..versionadded to the grains.  I'll get that in for you.

**Comment by toddtomkinson** [comment]
@cro, thanks.  For reference, here is a Vagrant project that demos how these can be used:

https://github.com/toddtomkinson/salt-proxy-demo

**Comment by jfindlay** [comment]
Excellent, thanks @toddtomkinson.

**Pull request merged**